concept/answer/ConceptMap.java

```diff
@@ -169,90 +169,110 @@ public int hashCode() {
 
     public static class Explainables {
 
-        Map<Retrievable, Explainable> explainableConcepts;
-        Map<Pair<Retrievable, Retrievable>, Explainable> explainableAttributeOwnerships;
+        private final Map<Retrievable, Explainable> explainableConcepts;
+        private final Map<Pair<Retrievable, Retrievable>, Explainable> explainableOwnerships;
+        private final Map<Retrievable, ? extends Concept> concepts;
 
-        public Explainables() {
-            this(unmodifiableMap(new HashMap<>()), unmodifiableMap(new HashMap<>()));
+        public Explainables(Map<Retrievable, ? extends Concept> concepts) {
+            this(unmodifiableMap(new HashMap<>()), unmodifiableMap(new HashMap<>()), concepts);
         }
 
         public Explainables(Map<Retrievable, Explainable> explainableConcepts,
-                            Map<Pair<Retrievable, Retrievable>, Explainable> explainableAttributeOwnerships) {
-            // to save memory, we only initialise these maps when we have to (we create a lot of concept maps)
+                            Map<Pair<Retrievable, Retrievable>, Explainable> explainableOwnerships,
+                            Map<Retrievable, ? extends Concept> concepts) {
             this.explainableConcepts = explainableConcepts;
-            this.explainableAttributeOwnerships = explainableAttributeOwnerships;
+            this.explainableOwnerships = explainableOwnerships;
+            this.concepts = concepts;
         }
 
         public FunctionalIterator<Explainable> explainables() {
-            return link(iterate(explainableConcepts.values()), iterate(explainableAttributeOwnerships.values()));
+            return link(iterate(explainableConcepts.values()), iterate(explainableOwnerships.values()));
```

Comment:
Hmmm. Method `explainables` in class `Explainables` seems ugly and indeed in the tests it results in calls to `.explainables().explainables()`.

How about we name this method `iterator`?

Reply:
sure